code/obj/machinery/computer/cloning.dm

```diff
@@ -887,6 +652,173 @@
 				set_lock(0)
 				active_process = PROCESS_IDLE
 
+/obj/machinery/computer/cloning/ui_act(action, params)
+	if(..())
+		return
+	switch(action)
+		if("delete")
+			var/selected_record =	find_record(params["ckey"])
+			if(selected_record)
+				logTheThing("combat", usr, null, "deletes the cloning record [selected_record["fields"]["name"]] for player [selected_record["fields"]["ckey"]] at [log_loc(src)].")
+				src.records.Remove(selected_record)
+				qdel(selected_record)
+				selected_record = null
+				src.temp = "Record deleted."
+				. = TRUE
+		if("scan")
+			if(!isnull(src.scanner))
+				src.scan_mob(src.scanner.occupant)
+				. = TRUE
+		if("clone")
+			var/ckey = params["ckey"]
+			if(ckey)
+				clone_record(find_record(ckey))
+				. = TRUE
+		if("toggleGeneticAnalysis")
+			if(pod1 && !pod1.operating)
+				pod1.gen_analysis = !pod1.gen_analysis
+				. = TRUE
+			else
+				src.temp = "Cannot toggle any modules while cloner is active."
+				. = TRUE
+		if("saveToDisk")
+			var/ckey = params["ckey"]
+			var/selected_record = find_record(ckey)
+			if ((isnull(src.diskette)) || (src.diskette.read_only) || (isnull(selected_record)))
+				src.temp = "Save error."
+				return
+
+			for (var/datum/computer/file/clone/R in src.diskette.root.contents)
+				if (R.fields["ckey"] == selected_record["fields"]["ckey"])
+					src.temp = "Record already exists on disk."
+					return
+
+			var/datum/computer/file/clone/cloneFile = new
+			cloneFile.name = "CloneRecord-[ckey(selected_record["fields"]["name"])]"
+			cloneFile.fields = selected_record["fields"]
+			src.temp = src.diskette.root.add_file(cloneFile) ? "Save successful." : "Save error."
+			. = TRUE
+
+		if("eject")
+			if (!isnull(src.diskette))
+				src.diskette.set_loc(src.loc)
+				usr.put_in_hand_or_eject(src.diskette) // try to eject it into the users hand, if we can
+				src.diskette = null
+				. = TRUE
+		if("load")
+			if (src.diskette.read_only)
+				// The file needs to be deleted from the disk after loading the record
+				src.temp = "Load error - cannot transfer clone records from a disk in read only mode."
+				return
+
+			var/loaded = 0
+
+			for(var/datum/computer/file/clone/cloneRecord in src.diskette.root.contents)
+				if (!find_record(cloneRecord.fields["ckey"]))
+					var/datum/data/record/R = new
+					R.fields = cloneRecord.fields
+					src.records += R
+					loaded++
+					src.temp = "Load successful, [loaded] [loaded > 1 ? "records" : "record"] transferred."
+					src.diskette.root.remove_file(cloneRecord)
+					. = TRUE
+
+			if(!loaded)
+				src.temp = "Load error."
+				return
```

Comment:
```suggestion
				return TRUE
```
even though this is returning early it's still changing the UI so we should probably still return TRUE